homeassistant/components/daikin/__init__.py

```diff
@@ -147,15 +149,36 @@ def _update_unique_id(entity_entry: er.RegistryEntry) -> dict[str, str] | None:
     if new_unique_id == old_unique_id:
         return
 
+    duplicate = dev_reg.async_get_device(
+        connections={(CONNECTION_NETWORK_MAC, new_mac)}, identifiers=None
+    )
+
+    # Remove duplicated device
+    if duplicate is not None:
+        if config_entry.entry_id in duplicate.config_entries:
+            _LOGGER.debug(
+                "Removing duplicated device %s",
+                duplicate.name,
+            )
+
+            # The automatic cleanup in entity registry is scheduled as a task, remove
+            # the entities manually to avoid unique_id collision when the entities
+            # are migrated.
+            duplicate_entities = er.async_entries_for_device(
+                ent_reg, duplicate.id, True
+            )
+            for entity in duplicate_entities:
+                ent_reg.async_remove(entity.entity_id)
+
+            dev_reg.async_remove_device(duplicate.id)
```

Comment:
How do we now that no other integration has a connection to this device?

It's better to use `dev_reg.async_update_device` and pass the config entry id to remove from the device. The device will then be removed if the config entry was the only config entry for the device.